src/lib/statistics/cardinality_estimator.cpp

```diff
@@ -0,0 +1,911 @@
+          // No overlapping bins: No rows selected
+          selectivity = 0.0f;
+          return;
+        }
+
+        const auto cardinality = column_vs_column_histogram->total_count();
+        selectivity = input_table_statistics->row_count == 0 ? 0.0f : cardinality / input_table_statistics->row_count;
+
+        /**
+         * Write out the AttributeStatistics of the scanned columns
+         */
+        const auto column_statistics = std::make_shared<AttributeStatistics<ColumnDataType>>();
+        column_statistics->histogram = column_vs_column_histogram;
+        output_column_statistics[left_column_id] = column_statistics;
+        output_column_statistics[*right_column_id] = column_statistics;
+
+      } else if (predicate.value.type() == typeid(ParameterID)) {
+        /**
+         * Estimate ColumnVsPlaceholder
+         */
+
+        switch (predicate.predicate_condition) {
+          case PredicateCondition::Equals: {
+            const auto total_distinct_count =
+                std::max(scan_statistics_object->total_distinct_count(), HistogramCountType{1.0f});
+            selectivity = total_distinct_count > 0 ? 1.0f / total_distinct_count : 0.0f;
+          } break;
+
+          case PredicateCondition::NotEquals: {
+            const auto total_distinct_count =
+                std::max(scan_statistics_object->total_distinct_count(), HistogramCountType{1.0f});
+            selectivity = total_distinct_count > 0 ? (total_distinct_count - 1.0f) / total_distinct_count : 0.0f;
+          } break;
+
+          case PredicateCondition::LessThan:
+          case PredicateCondition::LessThanEquals:
+          case PredicateCondition::GreaterThan:
+          case PredicateCondition::GreaterThanEquals:
+          case PredicateCondition::BetweenInclusive:
+          case PredicateCondition::BetweenExclusive:
+          case PredicateCondition::BetweenLowerExclusive:
+          case PredicateCondition::BetweenUpperExclusive:
+          case PredicateCondition::In:
+          case PredicateCondition::NotIn:
+          case PredicateCondition::Like:
+          case PredicateCondition::NotLike:
+            // Lacking better options, assume a "magic" selectivity for >, >=, <, <=, ... Any number would be equally
+            // right and wrong here. In some examples, this seemed like a good guess ¯\_(ツ)_/¯
+            selectivity = 0.5f;
+            break;
+
+          case PredicateCondition::IsNull:
+          case PredicateCondition::IsNotNull:
+            Fail("IS (NOT) NULL predicates should not have a 'value' parameter.");
+        }
+
+      } else {
+        /**
+         * Estimate ColumnVsValue
+         */
+
+        Assert(predicate.value.type() == typeid(AllTypeVariant), "Expected AllTypeVariant");
+
+        // TODO(anybody) For (NOT) LIKE predicates that start with a wildcard, Histograms won't yield reasonable
+        //               results. Assume a magic selectivity for now
```

Comment:
I don't see this "start with a wildcard" part here. Any chance the comments here are out-of-date?

Reply:
Todo was out of date, referred to code previously removed.